### DISCUSSION TITLE
Is there triangulation feature ?

### DISCUSSION THREAD

**Discussion by ivanvach** [mKeRix/room-assistant#840] [Q&A]
Is there triangulation feature that I can't find or I'm just out of luck ?
My house is 3 story house and when I'm say 20 ft from a room-assistant Pi it could mean 3 things
a) I'm 20 ft away on the same floor b) I'm one floor befelow c) I'm one floor above
obviously not acceptable. 

Triangulation is needed to compare location readings from 3 or more instances and determine location. 

Question is - is something like that built-in room-assistant or do I need to set it up on the site ?

**Comment by stancho44** [comment]
You can try going to the room-assistant ipaddress:6415/entities webpage to find the distance away from the devices. The website is in json format. Hope this helps

**Reply by ivanvach** [reply]
This is good yo know however it doesn't answer the original question and doesn't help resolving the issue of 3 story house where vertical location does matter.

**Comment by mKeRix** [comment]
At the moment the assumption is that you have a device running room-assistant in each room you want to track. Triangulation was requested in #17, although as discussed in that thread it might not lead to the outcome you'd like to see. Instead, I would love to have a machine learning implementation like proposed in #507, but I don't think I'll have the time to build this soon either.

As always, I'm happy to accept contributions for this!
